### PR TITLE
#782 associate higher and lower caching tiers' stats with ContextManager

### PR DESCRIPTION
**Pull request by lorban**

### CODE REVIEW

**Comment by ljacomet** [comment]
That build failure is a Java 1.6 compile failure

File: impl/src/main/java/org/ehcache/impl/internal/store/tiering/CompoundCachingTier.java [hunk 1]
```diff
@@ -67,6 +68,9 @@ public void onInvalidation(final K key, final Store.ValueHolder<V> valueHolder)
         }
       }
     });
+
+    StatisticsManager.associate(higher).withParent(this);
+    StatisticsManager.associate(lower).withParent(this);
```
Comment (chrisdennis) [inline impl/src/main/java/org/ehcache/impl/internal/store/tiering/CompoundCachingTier.java]: Calling static method through a subclass reference... might be better to call ContextManager.associate(...)

**Comment by ljacomet** [comment]
Failure is racy test, already filed as #775

File: impl/src/main/java/org/ehcache/impl/internal/store/tiering/CompoundCachingTier.java (hunk 1 above)
Comment (lorban) [inline impl/src/main/java/org/ehcache/impl/internal/store/tiering/CompoundCachingTier.java]: StatisticsManager.associate() is used everywhere. Should I change all refs while I'm at it?

Comment (chrisdennis) [inline impl/src/main/java/org/ehcache/impl/internal/store/tiering/CompoundCachingTier.java]: I guess not then... the cat's out of the bag... go with convention.

**Pull request merged**